Add sso configuration docs

tapadipti opened the pull request:
Closes https://github.com/iterative/studio/issues/8773

tapadipti commented:
@EduardMaghakyan This is wip as screenshots are yet to be added. But pls review the overall content and comment if anything major is missing.

EduardMaghakyan commented on content/docs/studio/user-guide/single-sign-on.md:
```diff
@@ -0,0 +1,48 @@
+
+## Okta integration
+
+**Configure Okta in DVC Studio team settings**
```
maybe it makes sense to move this below the `Add integration with DVC Studio in Okta` section? to get require values for `signOnUrl`, `issuer` `publicCert` they would need to first configure Okta.

tapadipti replied:
Thanks @EduardMaghakyan 
Can you suggest how I can try out sso and get screenshots? Should I sign up for a free trial with Okta? And when I try to open the `/saml/consume` page for my team in dev, I get `404 Nothing found`.

tapadipti commented:
@EduardMaghakyan Can you suggest how I can try out sso and get screenshots? Should I sign up for a free trial with Okta? And when I try to open the `/saml/consume` page for my team in dev, I get `404 Nothing found`.

EduardMaghakyan commented:
> And when I try to open the /saml/consume page for my team in dev, I get 404 Nothing found.
>
Hmm... that shouldn't be the case. Did you have `/api` in the path e.g. `https://dev.iterative.ai/api/teams/demo-team-with-sso/saml/consume` this should show an error about missing RelayState, but not 404

> Can you suggest how I can try out sso and get screenshots? Should I sign up for a free trial with Okta?
> 
I tried on our https://dev.iterative.ai/ and seems the team level SSO is disabled. And I assume `TEAM_SSO_ENALBED` is set to `False`. We need to enable this to 
I think it's actually the best if you create a free Okta dev account and try to configure everything. I guess that might reveal some gaps in the documentation that we provided. (This is already possible and active on `dev`) 
Until we have the #8772 the only way to configure SSO for the team is to make an API request
```
mutation {
  toggleSso(teamName: "<TEAM_NAME>", ssoEnabled: true) {
    ok
  }
  createOrUpdateSsoConfig(
    teamName: "<TEAM_NAME>"
    signOnUrl: "<OKTA_SING_ON_URL>"
    issuer: "<OKTA_ISSUER>"
    publicCert: """
    -----BEGIN CERTIFICATE-----
    <OKTA_IDP_PUBLIC_CERT>
    -----END CERTIFICATE-----
    """
  ) {
    ok
    error
  }
}
```

You can use [this example](https://dev.iterative.ai/api/graphql#query=mutation%20%7B%0A%20%20toggleSso(teamName%3A%20%22A_TEAM%22%2C%20ssoEnabled%3A%20true)%20%7B%0A%20%20%20%20ok%0A%20%20%7D%0A%20%20createOrUpdateSsoConfig(%0A%20%20%20%20teamName%3A%20%22A_TEAM%22%0A%20%20%20%20signOnUrl%3A%20%22https%3A%2F%2Fdev-72505381.okta.com%2Fapp%2Fdev-72505381_ssoateam_1%2Fexkef54ljvxehkmR75d7%2Fsso%2Fsaml%22%0A%20%20%20%20issuer%3A%20%22http%3A%2F%2Fwww.okta.com%2Fexkef54ljvxehkmR75d7%22%0A%20%20%20%20publicCert%3A%20%22%22%22%0A%20%20%20%20-----BEGIN%20CERTIFICATE-----%0A%20%20%20%20MIIDqDCCApCgAwIBAgIGAYz63%2BqSMA0GCSqGSIb3DQEBCwUAMIGUMQswCQYDVQQGEwJVUzETMBEG%0A%20%20%20%20A1UECAwKQ2FsaWZvcm5pYTEWMBQGA1UEBwwNU2FuIEZyYW5jaXNjbzENMAsGA1UECgwET2t0YTEU%0A%20%20%20%20MBIGA1UECwwLU1NPUHJvdmlkZXIxFTATBgNVBAMMDGRldi03MjUwNTM4MTEcMBoGCSqGSIb3DQEJ%0A%20%20%20%20ARYNaW5mb0Bva3RhLmNvbTAeFw0yNDAxMTEyMzMyNDdaFw0zNDAxMTEyMzMzNDdaMIGUMQswCQYD%0A%20%20%20%20VQQGEwJVUzETMBEGA1UECAwKQ2FsaWZvcm5pYTEWMBQGA1UEBwwNU2FuIEZyYW5jaXNjbzENMAsG%0A%20%20%20%20A1UECgwET2t0YTEUMBIGA1UECwwLU1NPUHJvdmlkZXIxFTATBgNVBAMMDGRldi03MjUwNTM4MTEc%0A%20%20%20%20MBoGCSqGSIb3DQEJARYNaW5mb0Bva3RhLmNvbTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoC%0A%20%20%20%20ggEBAJUhnIF3p%2FFPPDjf4FFIGzfoh01T9VpM%2FgmdIFCU1SX0WnP0H3Cyl%2BNYPPNF7v788KKUjmBb%0A%20%20%20%20xB4DH%2FX3J3MRCWURkYsHI%2FzYw08wQo6ZVPxO00UodH6quMHxBsiMlPXSMb9J9ivMV7HPnV%2F0Dn6R%0A%20%20%20%20KtqBVtdBSOjVPMV0KALZHv2si0vfVhWSykVGNTrB8V12Ce6Kd4TFMfl%2BBvgk53YE14KXwxEmzEoV%0A%20%20%20%201tHICI0HlgzQBqujJn%2FsZ4flFN8ym2Xh6GCKrau5sPdY%2FzsZU46Z3qlO%2FoAw9vsZ5gDCbYOcjPKR%0A%20%20%20%20K5D2j6idy1MBba6%2Ffp7xOGRYCjKSjTs7iL4tkL2XWGECAwEAATANBgkqhkiG9w0BAQsFAAOCAQEA%0A%20%20%20%20JOczuvVaFxuYISy0p0nuhMsQjBsNczQnw1o%2B%2F3kW4HbwesvsFIGxh%2FNexKEa8eOQU%2F3EVC%2Fvh1Vy%0A%20%20%20%20uJVEu75SOrUHac%2BrSVeG1NaIHGZN89DUrre5%2FIj1vv%2B13JG68fEkFD6BRtoHIELUiRbzPf41C9jR%0A%20%20%20%20HdcUl0iEms%2BdRneE2D95GpUJRJ5MFLSSqDSCLZmQEa%2FEzUe4NNrYHa8SgsW5DHOOzIfdsy27u%2FLp%0A%20%20%20%20qDI83ymbiV8agKzFJpqjOJHJHwK6nH44VkKnquw9cMGrdUN%2FAi4NPwg20plAL5fKk0eA2RJWJvik%0A%20%20%20%20TyY2Gs%2B%2FEyUHm3zaQLNsjMCMvERCfF1bzGNfWQ%3D%3D%0A%20%20%20%20-----END%20CERTIFICATE-----%0A%20%20%20%20%22%22%22%0A%20%20)%20%7B%0A%20%20%20%20ok%0A%20%20%20%20error%0A%20%20%7D%0A%7D)

Alternativley I can configure everything, add for you a user in Okta and you should be able to join my team.

EduardMaghakyan commented:
looks great @tapadipti . I belive everything is correct. Maybe we could mention that at the moment all the users who first time access the studio via SSO will be added to team with "Viewer" role?
If the roles need to be adjusted it has to be done manually on studio side atm.

EduardMaghakyan commented on content/docs/studio/user-guide/single-sign-on.md:
```diff
@@ -21,30 +21,71 @@ will be auto-assigned the [`Viewer` role](#roles).
+   If a user does not have a pre-assigned role when they sign in to a team, they
+   will be auto-assigned the `Viewer` role. If the role needs to be changed, it
+   has to be done in the `Collaborators` page in the DVC Studio team settings.
+
```
cool, thanks for adding this @tapadipti

tapadipti commented:
I've added the screenshots also. The PR is ready for final review.

dberenbaum commented:
Is it still waiting for any PRs in Studio? I see the `wait-core-merge` label but it's not clear if it's still relevant.

shcheklein commented on content/docs/studio/user-guide/team-collaboration/index.md:
```diff
@@ -162,6 +164,24 @@ connection to this server. Once you set up the connection, all your team members
 can connect to the Git repositories on this server. For more details, refer to
 [Custom GitLab Server Connection](/doc/studio/user-guide/git-integrations/custom-gitlab-server).
 
+### Configure Single Sign-on (SSO)
```
should it be coming before Okta? should we make Okta on a separate page?

shcheklein commented on content/docs/studio/user-guide/single-sign-on.md:
```diff
@@ -0,0 +1,98 @@
+
+SSO for teams can be configured by [team `admins`](#roles), and requires
+configuration on both DVC Studio and the IDP. The exact steps for this depend on
+the IDP.
```
IdP

shcheklein commented on content/docs/studio/user-guide/single-sign-on.md:
```diff
@@ -0,0 +1,98 @@
+
+1. **Create Enterprise account**: SSO is available for DVC Studio teams with
+   enterprise account subscriptions. If you are on the Free or Basic plan of DVC
+   Studio, [Contact us] to upgrade your account.
```
link is broken

shcheklein commented on content/docs/studio/user-guide/single-sign-on.md:
```diff
@@ -0,0 +1,98 @@
+   1. `Sign-on URL`: Paste the `Sign on URL` you copied from Okta
+   2. `Identity Provider Issuer URL`: Paste the `Issuer` you copied from Okta
+   3. `Public Certificate`: Paste the `Signing Certificate` you copied from
+      Okta. Remember to add `-----BEGIN CERTIFICATE-----` and
```
> Remember to add `-----BEGIN CERTIFICATE-----` ..

are those included when you copy it? can we make it more flexible? do we have a check on FE for this?

EduardMaghakyan replied:
we do have FE and BE checks. These are not available when you copy the certificate from Okta.
I had the same experience while configuring GitHub SAML

We could probably leave only the backend validation. Check like the certificate, in case of a validation error try to prepend and append the "BEGIN/END CERTIFICATE" and check again.

EduardMaghakyan replied:
I have added this issue to change the validation process https://github.com/iterative/studio/issues/9145

shcheklein commented on content/docs/studio/user-guide/single-sign-on.md:
```diff
@@ -0,0 +1,98 @@
+   have configured. For this, open Okta and follow these steps:
+
+   1. Open `Applications`
+   2. Click on `Assign Users to App`
```
will create a user? is it available already? cc @EduardMaghakyan

EduardMaghakyan replied:
@shcheklein yes we will create a user on our side once they log in the first time(I think it's called JIT provisioning).

shcheklein reviewed (approved):
great overall. We can probably mention that other provides work similarly cc @EduardMaghakyan to Okta and people can setup them as well? or what is your take - would it require modifications on BE to work with the other providers?

also some minor questions / checks

EduardMaghakyan commented:
>great overall. We can probably mention that other provides work similarly cc @EduardMaghakyan to Okta and people can setup them as well? or what is your take - would it require modifications on BE to work with the other providers?

@shcheklein I don't think we would require modifications.

maybe it's worth somehow emphasizing that we test and "officially" support Okta, but the other IdPs should also work?

shcheklein commented:
> great overall. We can probably mention that other provides work similarly cc @EduardMaghakyan to Okta and people can setup them as well? or what is your take - would it require modifications on BE to work with the other providers?

@tapadipti can we do the final step and we are good to merge this, after BEGIN / END is solved on prod we can update docs I think

tapadipti commented:
Thanks all for the review. I've addressed the comments.
This feature is not in prod yet - what are the issues that must be resolved before this can be enabled? (other than https://github.com/iterative/studio/issues/9145)

tapadipti commented:
@EduardMaghakyan What are the issues that must be resolved before this can be enabled in prod?

EduardMaghakyan commented:
@tapadipti I believe there are no pending issues related to this.
